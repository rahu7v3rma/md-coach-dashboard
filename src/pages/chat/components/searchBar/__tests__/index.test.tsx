import React from 'react';
import { act, create } from 'react-test-renderer';

import SearchBar from '../../searchBar';

describe('SearchBar component', () => {
    it('matches the snapshot', () => {
        const component = create(<SearchBar />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with a search icon and placeholder text', () => {
        const component = create(<SearchBar />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('updates the input value when typing', () => {
        const component = create(<SearchBar />);
        const instance = component.root;
        const inputElement = instance.findByProps({ name: 'groupname' });

        act(() => {
            inputElement.props.onChange({ target: { value: 'New Text' } });
        });

        expect(inputElement.props.value).toBe('New Text');
    });

    it('calls the onSearch callback with the search text when typing', () => {
        const onSearchMock = jest.fn();
        const component = create(<SearchBar onSearch={onSearchMock} />);
        const instance = component.root;
        const inputElement = instance.findByProps({ name: 'groupname' });

        act(() => {
            inputElement.props.onChange({ target: { value: 'New Text' } });
        });

        expect(inputElement.props.value).toBe('New Text');
        expect(onSearchMock).toHaveBeenCalledWith('New Text');
    });

    it('does not call onSearch callback when onSearch is not provided', () => {
        const onSearchMock = jest.fn();
        const component = create(<SearchBar />);
        const instance = component.root;
        const inputElement = instance.findByProps({ name: 'groupname' });

        act(() => {
            inputElement.props.onChange({ target: { value: 'New Text' } });
        });

        expect(inputElement.props.value).toBe('New Text');
        expect(onSearchMock).not.toHaveBeenCalled();
    });

    it('renders the component with an initial search text', () => {
        const component = create(<SearchBar />);
        const instance = component.root;
        const inputElement = instance.findByProps({ name: 'groupname' });
        expect(inputElement.props.value).toBe('');
    });
});
