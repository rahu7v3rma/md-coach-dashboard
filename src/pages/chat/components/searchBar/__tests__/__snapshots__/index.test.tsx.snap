// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component matches the snapshot 1`] = `
<div
  className="sc-bgqQcB eBGUbM"
>
  <img
    alt="search"
    className="sc-gTRrQi iZnMqf"
    src={Object {}}
  />
  <input
    className="sc-ewnqHT cFPklM"
    name="groupname"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;

exports[`SearchBar component renders with a search icon and placeholder text 1`] = `
<div
  className="sc-bgqQcB eBGUbM"
>
  <img
    alt="search"
    className="sc-gTRrQi iZnMqf"
    src={Object {}}
  />
  <input
    className="sc-ewnqHT cFPklM"
    name="groupname"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
